### ISSUE TITLE
All python kernelspecs installed to global locations are hidden when python extension is installed

### ISSUE THREAD

**Issue by IanMatthewHuff** [microsoft/vscode-jupyter#6389]
Seems to have been from the kernel search refactor changes. You can install kernelspecs all in the same interpreter. They could have different launch args or different environments. Syncing with the recent changes these kernelspecs are all hidden and unavailable in the kernel picker.

Example install:
![image.png](https://images.zenhubusercontent.com/5bdcd40a0faa800c749bac98/7c3556c1-a36b-48ed-8960-ff4e4fcf6412)

Listed by Jupyter:
![image.png](https://images.zenhubusercontent.com/5bdcd40a0faa800c749bac98/74906214-f4a9-4b0f-9bb3-2b0c2bf457c4)

Not listed in VS Code:
![image.png](https://images.zenhubusercontent.com/5bdcd40a0faa800c749bac98/041aeebf-292b-412e-b13e-de2c24fb620f)

Note: I was just testing this before pulling the changes so I'm sure they used to be listed.

**Comment by IanMatthewHuff** [comment]
We chatted a bit about this on teams quick. Feel like there is more here than we would rather deal with in release.

Issues for discussion:
1. Currently global python kernelspecs are hidden if you have the python extension installed. This is not right as you could have a globally installed python kernelspec that specifically customizes the environment. This previously was available, but is now hidden. Perhaps not super likely, but I have one of these that I use.
2. The current code calls out not returning global python kernelspecs as they are ambiguous as to the python environment to run them. 
3. However if you create a named kernelspec from --user (https://ipython.readthedocs.io/en/stable/install/kernel_install.html#kernels-for-different-environments) it will install globally but have the full interpreter path in the argv. Our current code can match this up with existing interpreters so the environment is known. 
4. Correct thing here I believe would be not to even care about if it's a default kernel spec or not, but instead to return all global python kernelspecs that can be identified to a python environment that we have.
5. Second point to check (and a concern when pulling in global pythons) it looks like when we find a kernelspec we hide any interpreters that match the path. If you have a global kernelspec with a custom environment you would not want to hide the interpreter as it's distinct from the interpreter: https://github.com/microsoft/vscode-jupyter/blob/617fccc68807d9055bf42fa26f05b3f96636cbb2/src/client/datascience/kernel-launcher/localPythonAndRelatedNonPythonKernelSpecFinder.ts#L152

**Comment by greazer** [comment]
This issue has to do with rethinking how we present kernels to users. So moving to October and marking as iteration-candidate, since I think there is more here overall.

**Comment by IanMatthewHuff** [comment]
@greazer I was actually just taking a peek at this when you moved it to October, do you mind if a put in a bit of time here still? I believe this particular issue is a straight up bug with the current implementation and might be a small fix.

**Comment by DonJayamanne** [comment]
@IanMatthewHuff  Can you still repro this issue?

**Comment by IanMatthewHuff** [comment]
@DonJayamanne Yup, I've got the repro and I'm debugging it this morning.

**Comment by IanMatthewHuff** [comment]
@connor4312 Thanks. Appreciate the extra work on this. Kernels can be pretty fussy to work with, especially for a funky setup like this. 

A couple of things on this:
Per the screen shot you posted it looks like you might have created the two kernel specs with the same name entry:
![image](https://user-images.githubusercontent.com/812783/139482920-fca52653-9c23-420b-8828-e2ab92ed7d6a.png)
Note that the --name is the same in both entries. You would want this to be different for each kernelspec so that two distinct kernelspecs are created. 

For the repro here the steps that are in the test plan item that I linked (https://github.com/microsoft/vscode/issues/133902) would be the validation steps. After a kernelspec is created using the command line arguments above it actually would not show up in the kernel picker list. At that point it's the exact same as the python environment that created it. It would only show up in our kernel picker list after you'd opened up the kernel.json file and edited the environment (steps for this were in the test plan item).

I did just check in a fix last night for this so making sure to have today's build of the Jupyter extension would help. But after you created the kernelspecs from the command line and edited the ENV block for them they would be expected to show up with the --display-name that you created for them. But only after the ENV block was changed and then after a window reload (we only look for new kernelspecs on activation).

I can also help on Slack / Teams today.
